packages/vats/src/crc.js

```diff
@@ -0,0 +1,176 @@
+/*
+ * https://en.wikipedia.org/wiki/Cyclic_redundancy_check
+ * Copyright (c) 2018 Vladimir Latyshev
+ * License: MIT
+ * Algorithms ported from https://pycrc.org/
+ * Forked from npm polycrc@1.1.0 for module system compatibility, then trimmed
+ * down to a version that assumes TypedArrays are universal.
```

Comment:
Embedding CRC6 is a reversible expedient. The existing `polycrc` doesn’t work with Endo because of a bad interaction with named exports. I may be able to fix that with more work on the Endo side. Otherwise, bringing this in-house and getting it past lint and tsc increased my confidence in the code, so there’s an upside. The downside is that I don’t think it’s worthwhile to also bring in the tests from `polycrc`. So, if this isn’t palatable, the alternatives are:

- Do more work on Endo until the CommonJS/ESM boundary works for this particular case. That is, an ESM importing a CJS with the previously working incantation `import { models as crcmodels } from 'polycrc'`, which requires detecting that `models` is an export detected by `module.exports = { models }` (approximately) in the lexical analysis.
- Patch `polycrc` so it’s `"type": "module"` and uses the appropriate ESM gestures.